### ISSUE TITLE
IntEnum __format__ behavior can't be overridden through __str__

### ISSUE THREAD

**Issue by f1292f21-f7dd-444b-a4f8-48407a510f53** [python/cpython#81660]
BPO | [37479](https://bugs.python.org/issue37479)
--- | :---
Nosy | @ericvsmith, @ethanfurman, @thatneat
PRs | <li>python/cpython#14545</li><li>python/cpython#22227</li><li>python/cpython#22228</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/ethanfurman'
closed_at = <Date 2020-09-13.20:51:07.553>
created_at = <Date 2019-07-01.23:24:11.546>
labels = ['3.8', 'type-bug', 'library', '3.9']
title = "IntEnum __format__ behavior can't be overridden through __str__"
updated_at = <Date 2020-09-13.20:51:07.551>
user = 'https://github.com/thatneat'
```

bugs.python.org fields:
```python
activity = <Date 2020-09-13.20:51:07.551>
actor = 'ethan.furman'
assignee = 'ethan.furman'
closed = True
closed_date = <Date 2020-09-13.20:51:07.553>
closer = 'ethan.furman'
components = ['Library (Lib)']
creation = <Date 2019-07-01.23:24:11.546>
creator = 'jason.curtis'
dependencies = []
files = []
hgrepos = []
issue_num = 37479
keywords = ['patch']
message_count = 7.0
messages = ['347089', '347090', '347094', '347096', '347291', '376848', '376851']
nosy_count = 3.0
nosy_names = ['eric.smith', 'ethan.furman', 'jason.curtis']
pr_nums = ['14545', '22227', '22228']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue37479'
versions = ['Python 3.8', 'Python 3.9']
```

</p></details>

**Comment by zinnjonas** [comment]
Was this change really intended?

The change will make projects break in their behavior when changed from 3.7 to a newer version.

`
from enum import IntEnum

class Test(IntEnum):
     A = 0
     B = 0

t = Test.A
print(f"0x{t:X} {str(t)}")
`

is valid python code and also compatible with python 3.7.
But!!!

`
from enum import IntEnum

class Test(IntEnum):
     A = 0
     B = 0
    
     def __str__(self):
             return self.name.lower()

t = Test.A
print(f"0x{t:X} {str(t)}")
`
is valid in python 3.7 but due to the change not valid in python 3.8+ anymore.
Since "str" can't be represented by "{:X}".
Since isinstance(t, int) is true in both scenarios, it is to be expected that f"0x{t:X}" will hold. it is an "int" and not a "str"

The __str__ function exists in both cases, the override of the second case shall not alter the behavior of the class !

**Comment by ethanfurman** [comment]
@zinnjonas This has been corrected in 3.11.
